### PR TITLE
ui: DestinationInputDirect component

### PR DESCRIPTION
**Pull request by mastercactapus**
**Description:**
This PR introduces the DestinationInputDirect component which will be used as a generic component for non-search-selectable destinations (e.g., a phone number or email address)

**Which issue(s) this PR fixes:**
Part of #2639

**Out of Scope:**
- The `DestinationField` meta-component will be in a separate PR.

**Screenshots:**
image
image

**Describe any introduced user-facing changes:**
N/A

**Describe any introduced API changes:**
N/A

**Additional Info:**
Run `make storybook` to interact with the new components.

### CODE REVIEW

File: web/src/app/selection/DestinationInputDirect.tsx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+  // fetch validation
+  const valid = !!data?.destinationFieldValidate
+
+  let adorn
+  if (!props.value || !props.supportsValidation) {
+    // no adornment if empty
+  } else if (valid) {
+    adorn = <Check className={classes.valid} />
+  } else if (valid === false) {
+    adorn = <Close className={classes.invalid} />
```
Comment (Forfold) [inline web/src/app/selection/DestinationInputDirect.tsx:93]:
```suggestion
    adorn = <Check sx={{ fill: green[500] }} />
  } else if (valid === false) {
    adorn = <Close sx={{ fill: red[500] }} />
```

File: web/src/app/selection/DestinationInputDirect.tsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+  }
+
+  let iprops: Partial<InputProps> = {}
+
+  if (props.prefix) {
+    iprops.startAdornment = (
+      <InputAdornment position='start' style={{ marginBottom: '0.1em' }}>
```
Comment (Forfold) [inline web/src/app/selection/DestinationInputDirect.tsx:100]:
```suggestion
      <InputAdornment position='start' sx={{ marginBottom: '0.1em' }}>
```

File: web/src/app/selection/DestinationInputDirect.tsx (hunk 1 above)
Comment (Forfold) [inline web/src/app/selection/DestinationInputDirect.tsx:93]: Let's use the sx prop instead, and import green & red from material to stay consistent

```suggestion
    adorn = <Check sx={{ fill: green[500] }} />
  } else if (valid === false) {
    adorn = <Close sx={{ fill: red[500] }} />
```

File: web/src/app/selection/DestinationInputDirect.tsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+    if (props.inputType === 'tel') {
+      e.target.value = e.target.value.replace(/[^0-9+]/g, '')
+    }
+
+    e.target.value = props.prefix + e.target.value
+    return props.onChange(e)
+  }
+
+  // TODO: what to do with input limiting (e.g., only allow digits)
```
Comment (Forfold) [inline web/src/app/selection/DestinationInputDirect.tsx:127]: still needed? out of scope?

**Review by Forfold — CHANGES_REQUESTED** [review]

File: web/src/app/selection/DestinationInputDirect.tsx (hunk 3 above)
Comment (mastercactapus) [reply web/src/app/selection/DestinationInputDirect.tsx:127]: good catch, we fixed this and forgot to remove the comment

**Review by tony-tvu — APPROVED** [review]

**Review by Forfold — APPROVED** [review]

**Pull request merged**